runtime/vm/classloadersearch.c

```diff
@@ -95,12 +93,43 @@ addToSystemClassLoaderSearch(J9JavaVM * vm, const char* pathSegment,
 		UDATA classLoaderType, BOOLEAN enforceJarRestriction)
 {
 	UDATA rc = CLS_ERROR_NONE;
+#ifdef WIN32
+	char *jarPath = NULL;
+	int32_t size = 0;
+	BOOLEAN conversionSucceed = FALSE;
+	
+	PORT_ACCESS_FROM_JAVAVM(vm);
+#endif /* WIN32 */
 
 	Trc_VM_addToSystemClassLoaderSearch_Entry(pathSegment);
 
-	if (pathSegment == NULL) {
-		return CLS_ERROR_NULL_POINTER;
+	if (NULL == pathSegment) {
+		rc = CLS_ERROR_NULL_POINTER;
+		goto done;
+	}
+
+#ifdef WIN32
```

Comment:
please use `#if defined(WIN32)` here and elsewhere

Reply:
Updated, please have another look.